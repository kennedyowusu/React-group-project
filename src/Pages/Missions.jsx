import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchMissions, selectMission } from '../redux/missions/missions';
import MissionsTable from '../components/Mission';

export default function Books() {
  const missionData = useSelector(selectMission);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missionData.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missionData]);

  return (
    <div className="mission-container">
      <MissionsTable missions={missionData} />
    </div>
  );
}
