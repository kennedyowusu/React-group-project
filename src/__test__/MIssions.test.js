import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Missions from '../components/Mission';

describe('Test Missions Component', () => {
  test('rockets component should match with snapshot', () => {
    const rockets = renderer
      .create(
        <Provider store={configureStore}>
          <Missions />
        </Provider>,
      )
      .toJSON();

    expect(rockets).toMatchSnapshot();
  });
});
