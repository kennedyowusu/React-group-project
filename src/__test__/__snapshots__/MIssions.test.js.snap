// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions Component rockets component should match with snapshot 1`] = `
<p>
  Loading...
</p>
`;
